Siv3D/src/Siv3D/Image/SivImage.cpp

```diff
@@ -1194,6 +1212,31 @@ namespace s3d
 		return rotated90();
 	}
 
+	Image Image::rotated90(int32 n) const&
+	{
+		switch (n % 4) // 時計回りに何回 90° 回転するか
+		{
+		case 1:
+		case -3:
+			return rotated90(); // 1 回または -3 回
+		case 2:
+		case -2:
+			return rotated180(); // 2 回または -2 回
+		case 3:
+		case -1:
+			return rotated270(); // 3 回または -1 回
+		default:
+			return *this; // 0 回
+		}
+	}
+
+	Image Image::rotated90(int32 n)&&
+	{
+		// rotate90() が最適化されたら次の実装に変更する
+		// return std::move(rotate90(n));
+		return rotated90(n);
```

Comment:
```suggestion
		switch (n % 4) // 時計回りに何回 90° 回転するか
		{
		case 1:
		case -3:
			return std::move(*this).rotated90(); // 1 回または -3 回
		case 2:
		case -2:
			return std::move(*this).rotated180(); // 2 回または -2 回
		case 3:
		case -1:
			return std::move(*this).rotated270(); // 3 回または -1 回
		default:
			return std::move(*this); // 0 回
		}
```

`Image::rotated90() &&` では、 `Image::rotate90()` が最適化されておらず `Image::rotated90() const&` のほうが高速であるため `Image::rotated90() const&` に委譲する実装にしてありますが、`Image::rotated180() &&` についてはそうではありません。
また、この実装では0回回転する場合にも不要なコピーが発生してしまいます。

ここは、`Image::rotated90(int32) const&` に委譲するのではなく、回転0回のときは自身をムーブし、それ以外のときはそれぞれの `Image::rotated●●() &&` を呼び出すように実装するほうが良いです。